srml/contract/src/exec.rs

```diff
@@ -625,6 +634,15 @@ where
 	fn deposit_event(&mut self, data: Vec<u8>) {
 		self.ctx.events.push(RawEvent::Contract(self.ctx.self_account.clone(), data));
 	}
+
+	fn set_rent_allowance(&mut self, rent_allowance: BalanceOf<T>) {
+		self.ctx.overlay.set_rent_allowance(&self.ctx.self_account, rent_allowance)
+	}
+
+	fn rent_allowance(&self) -> BalanceOf<T> {
+		self.ctx.overlay.get_rent_allowance(&self.ctx.self_account)
+			.unwrap_or(<BalanceOf<T>>::zero()) // Must never be triggered actually
```

Comment:
If this must never be triggered, this should be
```suggestion
			.expect("Rent allowance must never be `None`")
```

The only reason I can think for not panicking is if fetching rent can fail with an I/O error.

Reply:
the runtime can never panic.

Reply:
> With all panickers throughout all of Parity's codebases, it's PROVE or REMOVE. You either have to **prove** that the panicker can not possibly ever happen, or remove it and continue in some other way. This isn't a proof, therefore it must offer some sort of alternative way forward. A vaguely sensible default is fine here.

Is “log a fatal error and exit” acceptable in some cases?  I am specifically thinking about a bad genesis block, though there are probably others.